src/test/kotlin/org/openrewrite/java/migrate/guava/NoGuavaCreateTempDirTest.kt

```diff
@@ -0,0 +1,235 @@
+package org.openrewrite.java.migrate.guava
+
+import org.junit.jupiter.api.Test
+import org.openrewrite.Recipe
+import org.openrewrite.java.JavaParser
+import org.openrewrite.java.JavaRecipeTest
+
+@Suppress("ResultOfMethodCallIgnored", "UnstableApiUsage")
+class NoGuavaCreateTempDirTest : JavaRecipeTest {
+    override val parser: JavaParser
+        get() = JavaParser.fromJavaVersion()
+            .logCompilationWarningsAndErrors(true)
+            .classpath("guava")
+            .build()
+
+    override val recipe: Recipe
+        get() = NoGuavaCreateTempDir()
+
+    @Test
+    fun inMethodThrowingIoException() = assertChanged(
+        before = """
+            import java.io.File;
+            import java.io.IOException;
+            import com.google.common.io.Files;
+            
+            class A {
+                void doSomething() throws IOException {
+                    File dir = Files.createTempDir();
+                    dir.createNewFile();
+                }
+            }
+        """,
+        after = """
+            import java.io.File;
+            import java.io.IOException;
+            import java.nio.file.Files;
+            
+            class A {
+                void doSomething() throws IOException {
+                    File dir = Files.createTempDirectory(null).toFile();
+                    dir.createNewFile();
+                }
+            }
+        """
+    )
+
+    @Test
+    fun tempDirIsFieldVar() = assertChanged(
+        before = """
+            import java.io.File;
+            import com.google.common.io.Files;
+            
+            class A {
+                File gDir = Files.createTempDir();
+            }
+        """,
+        after = """
+            import java.io.File;
+            import java.io.IOException;
+            import java.nio.file.Files;
+            
+            class A {
+                File gDir = createTempDirectory();
+            
+                private static File createTempDirectory() {
```

Comment:
I really don't like generating boilerplate methods like this in recipes. Maybe we limit the places where we apply the recipe to scopes that already return or otherwise catch an `IOException`.